test/test_util.py

```diff
@@ -255,10 +257,19 @@ def test_encrypt_decrypt():
         assert data == decrypted
 
 
-def test_detect_running_process_id():
-    pytest_id = myhoard_util.detect_running_process_id("python3 -m pytest")
-    coverage_id = myhoard_util.detect_running_process_id("python3 -m coverage")
-    if pytest_id is None and coverage_id is None:
-        output = subprocess.check_output(["ps", "-x", "--cols", "1000", "-o", "pid,command"])
-        raise AssertionError(f"None of the commands included in ps output {output!r}")
-    assert myhoard_util.detect_running_process_id("certainlynosuchprocesscurrentlyrunning") is None
+class TestDetectRunningProcessId:
```

Comment:
pytest does not recommend xunit-style test classes with setup/teardown functions, using fixtures is more native to the framework and composes better. (it's not possible to ask another fixture from the setup/teardown functions for instance).

Reply:
I suppose you're referring to this:

```While these setup/teardown methods are simple and familiar to those coming from a unittest or nose background, you may also consider using pytest’s more powerful [fixture mechanism](https://docs.pytest.org/en/6.2.x/fixture.html#fixture) which leverages the concept of dependency injection, allowing for a more modular and more scalable approach for managing test state, especially for larger projects and for functional testing. You can mix both fixture mechanisms in the same file but test methods of unittest.TestCase subclasses cannot receive fixture arguments.```

I'd not say it's not recommended, just that it's an alternative approach.

"more native to the framework"  -> doesn't mean better, though. What advantage would we get from this 
"composes better" -> we don't need composition here, it's a very small test, and I think I've seen plenty of examples where fixtures are abused and reused too many times, and their dependency chain can lead to an hard-to-identify test behaviour.

My take would be "use a fixture where a fixture is useful", not "use a fixture as the default approach". The test as written is very self-contained and quite easy to understand. I see no advantage in using a fixture with a cleanup action.